app/Core/Repository/ServiceRepository.cs

```diff
@@ -86,6 +86,76 @@ public Services GetService(Services service)
             return null;
         }
 
+        public bool IsDuplicateServiceCode(string serviceCode)
+        {
+            string query = "SELECT COUNT(*) FROM services WHERE serviceCode = @ServiceCode AND isDeleted = 0";
```

Comment:
Hi @lhyra13, based on the query - we are checking duplicate non-deleted service code in the service database. But how about those deleted service code? 

If the scenario is:

We have an existing Service Code = TEST but has isDeleted=1.
Then we add another Service Code with the same value as TEST but has isDeleted=0.
- Will this be called duplicate or not? 
- Should the client need to recreate a new code even it is already existed or should they set the deleted one to active to be use again?

Reply:
Hi @lhyra13, kindly check issues I encountered during my review.

1. **Saving new service record.**

    **UI Form (frmServicesModal.cs)**
    ![image](https://github.com/user-attachments/assets/f21abf06-5052-4446-9f9c-994700c16d72)

    **DB (vcms.services)**
    ![image](https://github.com/user-attachments/assets/9a300be4-5c82-4ba9-906f-689940fcef0a)

    **Issue:** Provided ServiceCode=SERVICECODE1 is not yet existed in services table, but the program tells that it is already existed.

4. **Update**
Updating the service code of a record doesn't reflect into the database. Also there's no checking if the updated service code is already used by other record. It should also have that kind of validation here in UPDATE action.
5. **Field validation**
I can enter a whitespace Service code, description, and price.
![image](https://github.com/user-attachments/assets/980cecf2-4f23-4e01-8902-5dbf36ceecc7)
**NOTE**: The `Price` field with value `1` in the image, has a `space` character before the numeric value `1`. I actually enter it like this ` 1`. 

**Regarding on your question:**
- **Q**: _If the existing code (with isDeleted = 1) should be used again, how can we set isDeleted = 0 to reactivate it?_
- **A**: Instead of `isDeleted`, we can implement a Status dropdown with value 'Active','Inactive' in the service record. Similar to what I implemented in the Account Module for User records. Additionally, if this will be the new approach we should also change the behavior of the `Remove` to set the selected record as 'Inactive' to remove.

Reply:
done with this issue , please review my changes . thank you 


I update my database , I add constraints in adding negative value in price.

ALTER TABLE services
ADD CONSTRAINT CHK_Price_NonNegative
CHECK (Price >= 0);